kazoo/protocol/connection.py

```diff
@@ -723,7 +723,7 @@ def _connect(self, host, port):
         else:
             self._set_connected_ro_or_rw(client)
             for scheme, auth in client_auth_data_copy:
-                if scheme == "digest":
+                if scheme != "sasl":
```

Comment:
I am not sure I understand what you are trying to achieve here.
There is no such thing as a "sasl" scheme for client auth data. The SASL negotiation is done outside of Zookeeper client auth data